onnxruntime/python/tools/symbolic_shape_infer.py

```diff
@@ -700,6 +702,22 @@ def _infer_Expand(self, node):
                 helper.make_tensor_value_info(node.output[0], self.known_vi_[node.input[0]].type.tensor_type.elem_type,
                                               new_shape))
 
+    def _infer_Transpose(self, node):
+        data_shape = self._get_shape(node, 0)
+        vi = self.known_vi_[node.output[0]]
+        perm = get_attribute(node, 'perm', reversed(list(range(len(data_shape)))))
+
+        new_sympy_shape = self._get_sympy_shape(node, 0)
+        for i, perm_idx in enumerate(perm):
+            new_sympy_shape[i] = data_shape[perm_idx]
```

Comment:
>new_sympy_shape[i] = data_shape [](start = 12, length = 31)

Why assign from shape to sympy_shape? self._get_shape returns a list of string or int, while self._get_sympy_shape returns list of sympy expression or literals.